lightningd/chaintopology.c

```diff
@@ -229,33 +229,40 @@ static void broadcast_done(struct bitcoind *bitcoind,
 	tal_del_destructor2(otx->channel, clear_otx_channel, otx);
 
 	if (otx->finished) {
-		otx->finished(otx->channel, success, msg);
-		tal_free(otx);
-	} else if (we_broadcast(bitcoind->ld->topology, &otx->txid)) {
+		if (otx->finished(otx->channel, otx->tx, success, msg, otx->cbarg)) {
+			tal_free(otx);
+			return;
+		}
+	}
+
+	if (we_broadcast(bitcoind->ld->topology, &otx->txid)) {
 		log_debug(
 		    bitcoind->ld->topology->log,
 		    "Not adding %s to list of outgoing transactions, already "
 		    "present",
 		    type_to_string(tmpctx, struct bitcoin_txid, &otx->txid));
 		tal_free(otx);
-	} else {
-		/* For continual rebroadcasting, until channel freed. */
-		tal_steal(otx->channel, otx);
-		outgoing_tx_map_add(bitcoind->ld->topology->outgoing_txs, otx);
-		tal_add_destructor2(otx, destroy_outgoing_tx, bitcoind->ld->topology);
+		return;
```

Comment:
had to double-check this is semantically the same as the previous if-else-statement.